### PR TITLE
Feature/baip 1331 initial setup OVS export

### PR DESCRIPTION
**Pull request by bennyvdhoogen**

### CODE REVIEW

File: src/api/span-installation-export/span-installation-export.module.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { Module } from '@nestjs/common';
+import { CqrsModule } from '@nestjs/cqrs';
+// import { ConsoleModule } from 'nestjs-console';
+// import { HttpModule } from '@nestjs/axios';
```
Comment (anaim) [inline src/api/span-installation-export/span-installation-export.module.ts:4]: Can be removed

File: src/api/span-installation-export/span-installation-export.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { QueryBus } from '@nestjs/cqrs';
+import { Response } from 'express';
+
+import { survey } from '../../schema/survey/__stubs__';
```
Comment (anaim) [inline src/api/span-installation-export/span-installation-export.controller.spec.ts:5]: Can be removed?

File: src/api/span-installation-export/span-installation-export.controller.spec.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+			set: jest.fn(),
+			send: jest.fn(),
+		};
+
+		const module: TestingModule = await Test.createTestingModule({
+			controllers: [SpanInstallationExportController],
+			providers: [{ provide: QueryBus, useValue: queryBusMock }],
+		}).compile();
+
+		// controller = module.get<SpanInstallationExportController>(SpanInstallationExportController);
```
Comment (anaim) [inline src/api/span-installation-export/span-installation-export.controller.spec.ts:30]: Can be removed?

File: src/api/span-installation-export/queries/export-batch-data.spec.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+			}
+		} as DateConstructor;
+	});
+
+	afterAll(() => {
+		global.Date = realDate;
+	});
+
+	beforeEach(() => {
+		// exporterServiceMock.getDummyData.mockClear();
```
Comment (anaim) [inline src/api/span-installation-export/queries/export-batch-data.spec.ts:39]: Can be removed?

File: src/api/span-installation-export/span-installation-export.repository.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+				id: true,
+				objectId: true,
+			},
+		});
+
+		const spanInstallations = await this.prisma.objects.findMany({
+			where: {
+				id: {
+					in: surveys.map((survey) => survey.objectId),
+				},
```
Comment (anaim) [inline src/api/span-installation-export/span-installation-export.repository.ts:28]: Can be removed?

File: src/schema/batch/batch.repository.ts [hunk 6]
```diff
@@ -3,12 +3,38 @@ import { Injectable } from '@nestjs/common';
...
+			where: {
+				id: batchId,
+			},
+		});
+
+		return batch;
+	}
+
+	async getAllOVSBatches(): Promise<DBBatch[]> {
+		// Wrote this awkward rawQuery as I could not figure out how to do this with the version of Prisma we use
```
Comment (fortunato) [inline src/schema/batch/batch.repository.ts:23]: Did you find this?
https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filtering-on-an-array-value

As it is, the type cast and assignment to batch are redundant I'd say. My IDE appears totally happy with this:
``` typescript
return this.prisma.$queryRaw`
	SELECT b.*, inspectionTypesText
	FROM "batches" b
	cross join lateral json_array_elements_text ( "inspectionStandardTypes" ) as inspectionTypesText
	WHERE
		inspectionTypesText like '%spanInstallation%'
		AND status = 'active';
`;
```

Comment (bennyvdhoogen) [reply src/schema/batch/batch.repository.ts:23]: I did stumble upon that yes, but I could not get `array_contains` as a filter to work. When I wrote this we were definitely not on Prisma v4 and I do recall that having to do with my decision to go for a raw query. See:   

> Filter on a Json field
From v2.23.0, you can filter rows by the data inside a Json type. We call this advanced Json filtering. The availability of advanced Json filtering depends on your Prisma version: V4.0.0 or later: advanced Json filtering is [generally available] (https://www.prisma.io/docs/about/prisma/releases#generally-available-ga).

(https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#:~:text=V4.0.0%20or%20later%3A%20advanced%20Json%20filtering%20is%20generally%20available.)

I see a dependabot PR is open to migrate to v4.x -- perhaps worth revisiting this method after that one has been merged?

File: src/api/span-installation-export/span-installation-export.repository.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+				name: true,
+				code: true,
+				location: true,
+				latitude: true,
+				longitude: true,
+				attributes: true,
+			},
+		});
+
+		const output = spanInstallations.map((spanInstallation) => {
```
[the remark this answers is not in the document]
Comment (bennyvdhoogen) [reply src/api/span-installation-export/span-installation-export.repository.ts:41]: Fixed; see PR > https://github.com/Amsterdam/bmi-aip-graphql-service/pull/459

File: src/api/span-installation-export/queries/export-all-data.spec.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+import { ExportAllDataHandler } from './export-all-data.handler';
+import { ExportAllDataQuery } from './export-all-data.query';
+
+const workbook = new ExcelJS.Workbook();
+const worksheet = workbook.addWorksheet('Mock');
+worksheet.addRow([]);
+
+const exporterServiceMock: MockedObjectDeep<SpanInstallationExportService> = {
+	// @ts-ignore
```
[the remark this answers is not in the document]
Comment (bennyvdhoogen) [reply src/api/span-installation-export/queries/export-all-data.spec.ts:14]: Indeed, removed; see PR > https://github.com/Amsterdam/bmi-aip-graphql-service/pull/459

**Pull request merged**